packages/backend/src/payment-pointer/payment-pointer.service.ts

```diff
@@ -0,0 +1,95 @@
+import { NextFunction, Request, Response } from 'express'
+import { BaseResponse } from '../shared/models/BaseResponse'
+import { zParse } from '../middlewares/validator'
+import { paymentPointerSchema } from './payment-pointer.schema'
+import { Account } from '../account/account.model'
+import { createRafikiPaymentPointer } from '../rafiki/request/payment-pointer.request'
+import { PaymentPointerModel } from './payment-pointer.model'
+import { getUserIdFromRequest } from '../utils/getUserId'
+
+export const createPaymentPointer = async (
+  req: Request,
+  res: Response<BaseResponse<PaymentPointerModel>>,
+  next: NextFunction
+) => {
+  try {
+    const userId = getUserIdFromRequest(req)
+    const accountId = req.params.accountId
+    const { publicName } = await zParse(paymentPointerSchema, req)
+    const account = await Account.query()
+      .findById(accountId)
+      .where('userId', userId)
+      .throwIfNotFound()
+
+    const rafikiPaymentPointer = await createRafikiPaymentPointer(
+      publicName,
+      account.assetRafikiId
+    )
+    const paymentPointer = await PaymentPointerModel.query().insert({
+      publicName,
+      accountId,
+      id: rafikiPaymentPointer.id
+    })
+
+    return res.json({
+      success: true,
+      message: 'Payment pointer created',
+      data: paymentPointer
+    })
+  } catch (e) {
+    next(e)
+  }
+}
+export const listPaymentPointers = async (
+  req: Request,
+  res: Response<BaseResponse<PaymentPointerModel[]>>,
+  next: NextFunction
+) => {
+  try {
+    const userId = getUserIdFromRequest(req)
+    const accountId = req.params.accountId
+    // Validate that account id belongs to current user
+    const account = await Account.query()
+      .findById(accountId)
+      .where('userId', userId)
+      .throwIfNotFound()
```

Comment:
Same here, I suggest checking the existance of the account after the query is being made, and throwing a custom error manually.
Maybe it's BadRequest or Forbidden(403)

```suggestion
      if(!account){
         throw new BadRequestException()
      }
```